Formula/ani-cli.rb

```diff
@@ -0,0 +1,18 @@
+class AniCli < Formula
+  desc "A cli tool to browse and play anime"
+  homepage "https://github.com/pystardust/ani-cli/"
+  sha256 "b53a8ebf7bf44bc7497ad4b4247e941ca9a7a89365717d3f4a2da349bcc80e45"
+  url "https://github.com/pystardust/ani-cli/archive/refs/tags/v1.6.tar.gz"
+  license "GPL-3.0"
+
+  depends_on "grep"
+  depends_on "curl"
+  depends_on "openssl"
+  depends_on "cask"
+  depends_on "aria2"
+  depends_on "mpv"
+  
+  def install
+    bin.install 'ani-cli'
+  end
+end
```

Comment:
We need a test that exercises the some of the functionality of the app. Version checks or usage checks (foo --version or foo --help) are not sufficient, as explained in the formula cookbook.

In most cases, a good test would involve running a simple test case: run #{bin}/foo input.txt.

- Then you can check that the output is as expected (with assert_equal or assert_match on the output of shell_output)
- You can also check that an output file was created, if that is expected: assert_predicate testpath/"output.txt", :exist?

Some advice for specific cases:

- If the formula is a library, compile and run some simple code that links against it. It could be taken from upstream's documentation / source examples.
- If the formula is for a GUI program, try to find some function that runs as command-line only, like a format conversion, reading or displaying a config file, etc.
- If the software cannot function without credentials, a test could be to try to connect with invalid credentials (or without credentials) and confirm that it fails as expected.
- Same if the software requires a virtual machine, docker instance, etc. to be running.

Reply:
The program is mostly interactive, acting similar to a curses based program.
We have very few non-interactive options, two of wich would be printing help and the version string.
The only other non-interactive option is `-a`, but since that one opens mpv to play an episode directly, you can't exactly assert it either.
Even if there was a function x that just outputs the extracted video url to STDOUT, we wouldn't be able to assert that since gogoanime always returns different URLs.

Provided this context, how can we comply with testing policy?

Reply:
> The program is mostly interactive, acting similar to a curses based program.

There are plenty of those that are tested in Homebrew. See for example `sfst`

>  Even if there was a function x that just outputs the extracted video url to STDOUT, we wouldn't be able to assert that since gogoanime always returns different URLs.

Doesn't it always contain some standard output? Like gogoanime.com or something?